src/v2/cookbook/creating-custom-scroll-directives.md

```diff
@@ -0,0 +1,152 @@
+---
+title: Creating Custom Scroll Directives
+type: cookbook
+order: 5
+---
+
+## Simple Example
+
+There are many times that we might want to add a bit of behavior, especially animation, to a scroll event on a site. There are many ways to do so, but the path with the least amount of code and dependencies is to use a [custom directive](https://vuejs.org/v2/guide/custom-directive.html) to create a hook for anything that fires off a particular scroll event.
+
+``` js
+Vue.directive('scroll', {
+  inserted: function(el, binding) {
+    let f = function(evt) {
+      if (binding.value(evt, el)) {
+        window.removeEventListener('scroll', f);
+      }
+    };
+    window.addEventListener('scroll', f);
+  },
+});
+
+// main app
+new Vue({
+  el: '#app',
+  methods: {
+   handleScroll: function(evt, el) {
+    if (window.scrollY > 50) {
+      el.setAttribute("style", "opacity: 1; transform: translate3d(0, -10px, 0)")
+    }
+    return window.scrollY > 100;
+    }
+  }
+});
+```
+
+``` html
+<div id="app">
+  <h1 class="centered">Scroll me</h1>
+  <div class="box" v-scroll="handleScroll">
+    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. A atque amet harum aut ab veritatis earum porro praesentium ut corporis. Quasi provident dolorem officia iure fugiat, eius mollitia sequi quisquam.</p>
+  </div>
+</div>
+```
+
+<p class="tip">Remember! The directive must be registered before the Vue instance</p>
+
+We'd also need a style property that will transition the intermediary values here, in this case: 
+
+```
+.box {
+  transition: 1.5s all cubic-bezier(0.39, 0.575, 0.565, 1);
+}
+```
+
+<p data-height="450" data-theme-id="5162" data-slug-hash="983220ed949ac670dff96bdcaf9d3338" data-default-tab="result" data-user="sdras" data-embed-version="2" data-pen-title="Custom Scroll Directive- CSS Transition" class="codepen">See the Pen <a href="https://codepen.io/sdras/pen/983220ed949ac670dff96bdcaf9d3338/">Custom Scroll Directive- CSS Transition</a> by Sarah Drasner (<a href="https://codepen.io/sdras">@sdras</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
+
+Or, with GreenSock(GSAP) or any other JavaScript animation library, the code becomes even more simple:
+
+``` js
+Vue.directive('scroll', {
+  inserted: function(el, binding) {
+    let f = function(evt) {
+      if (binding.value(evt, el)) {
+        window.removeEventListener('scroll', f);
+      }
+    };
+    window.addEventListener('scroll', f);
+  },
+});
+
+// main app
+new Vue({
+  el: '#app',
+  methods: {
+   handleScroll: function(evt, el) {
+    if (window.scrollY > 50) {
+      TweenMax.to(el, 1.5, {
+        y: -10,
+        opacity: 1,
+        ease: Sine.easeOut
+      })
+    }
+    return window.scrollY > 100;
```

Comment:
Same indent issue I guess.

Reply:
I should just run it through prettier. Too bad the default formatter for codepen is a different style than what we use so I can't use it. Thanks!